nni/nas/experiment/pytorch.py

```diff
@@ -303,7 +303,9 @@ def run(self,
                 else:
                     raise RuntimeError(f'The experiment mode "{self._action}" is not supposed to invoke run() method.')
 
-                self._run_strategy(base_model_ir, self.applied_mutators)
+                # non-block
+                self._run_strategy_thread = Thread(target=self._run_strategy, args=(base_model_ir, self.applied_mutators), daemon=True)
+                self._run_strategy_thread.start()
```

Comment:
some concerns here. 
Python cannot explicitly terminate a thread, so _run_strategy will still activate even Experiment.run() returned.

Reply:
Starting strategy in the main thread is by design. We did this for debuggability and potential needs for interactive experiments.
Also, as you mentioned, there could be potential risks that prevent strategy from terminating in tricky cases.

Overall, I think the correct way to do this is: the strategy could have a way to check whether experiment "should" stop, and terminate itself when it "should". This is already in a draft design, and might come in v3.0 or v3.1.

Reply:
Yes, I also think it would be better to let strategy decide when to terminate the search.

I'll close this pr. Look forward to your design :).